### ISSUE TITLE
Instrumentations based on legacy activity creation provide default Kind to sampler

### ISSUE THREAD

**Issue by cijothomas** [open-telemetry/opentelemetry-dotnet#1947]
Instrumentations for libraries like Asp.Net Core, Asp.Net, HttpClient (.NET Core) are somewhat special case (https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/docs/trace/extending-the-sdk#special-case--instrumentation-for-libraries-producing-legacy-activity). These instrumentations feed the legacy activity through sampler by leveraging `AddLegacySource`  capability in the TracerProvider. As these activities are sent to sampler *before* instrumentation itself gets run, the `ActivityKind` for them will be `Internal` (the default) when seen by the sampler.
After the sampler is executed, the instrumentations get the DS callback and the Kind will be correctly set then. 

In short, when Samplers receive legacy activities, the kind will be the default (Internal one).
One potential way of fixing this would be to enhance `AddLegacySource` to also support setting the Kind.

**Comment by vishweshbankwar** [comment]
@cijothomas - please assign this to me.

**Comment by vishweshbankwar** [comment]
@cijothomas - This will be fixed .NET7.0 onwards. Do we need to look into fixing for .NET6.0?

**Comment by cijothomas** [comment]
Closing this issue as I believe this is not worth the effort solving for older frameworks (fully aware .NET 6.0 is still supported). If disagreeing, please re-open. Thanks!
